### PR TITLE
Move readCollection, readCollections, and readDatabases to ARM

### PR DESCRIPTION
**Pull request by victor-meng**
Move readCollection, readCollections, and readDatabases to ARM

### CODE REVIEW

File: src/Common/dataAccess/deleteCollection.test.ts [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
+
+  it("should call ARM if logged in with AAD", async () => {
+    window.authType = AuthType.AAD;
     await deleteCollection("database", "collection");
     expect(armRequest).toHaveBeenCalled();
   });
-  // TODO: Test non-AAD case
+
+  it("should call SDK if not logged in with non-AAD method", async () => {
+    window.authType = AuthType.MasterKey;
```
Comment (southpolesteve) [inline src/Common/dataAccess/deleteCollection.test.ts:29]: Thanks!

File: src/Common/dataAccess/readCollection.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { client } from "../CosmosClient";
+import { logConsoleProgress, logConsoleError } from "../../Utils/NotificationConsoleUtils";
+import { logError } from "../Logger";
+import { sendNotificationForError } from "./sendNotificationForError";
+
+export async function readCollection(databaseId: string, collectionId: string): Promise<DataModels.Collection> {
+  let collection: DataModels.Collection;
+  const clearMessage = logConsoleProgress(`Querying container ${collectionId}`);
+  try {
+    const response = await client()
```
Comment (ato9000) [inline src/Common/dataAccess/readCollection.ts:11]: Where is the RP support for readCollection?

File: src/Common/dataAccess/readCollections.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import { logError } from "../Logger";
+import { sendNotificationForError } from "./sendNotificationForError";
+import { userContext } from "../../UserContext";
+
+export async function readCollections(databaseId: string): Promise<DataModels.Collection[]> {
+  let collections: DataModels.Collection[];
+  const clearMessage = logConsoleProgress(`Querying containers for database ${databaseId}`);
+  try {
+    if (window.authType === AuthType.AAD) {
+      const rpResponse = await listSqlContainers(
```
Comment (ato9000) [inline src/Common/dataAccess/readCollections.ts:15]: Is this for SQL only or does listSqlContainers takes care of all APIs?

File: src/Common/dataAccess/readDatabases.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import { logError } from "../Logger";
+import { sendNotificationForError } from "./sendNotificationForError";
+import { userContext } from "../../UserContext";
+
+export async function readDatabases(): Promise<DataModels.Database[]> {
+  let databases: DataModels.Database[];
+  const clearMessage = logConsoleProgress(`Querying databases`);
+  try {
+    if (window.authType === AuthType.AAD) {
+      const rpResponse = await listSqlDatabases(
```
Comment (ato9000) [inline src/Common/dataAccess/readDatabases.ts:15]: Same question: is this SQL only or does listSqlDatabases handles all APIs?

File: src/Common/dataAccess/readCollection.ts (hunk 2 above)
Comment (victor-meng) [reply src/Common/dataAccess/readCollection.ts:11]: As discussed offline, readCollection is only called during the resource token workflow so RP support is not required.

File: src/Common/dataAccess/readCollections.ts (hunk 3 above)
Comment (victor-meng) [reply src/Common/dataAccess/readCollections.ts]: I've added support for all APIs in the latest commit.

**Pull request merged**